diff --git a/installperl b/installperl
index e2006c5..cd19999 100755
--- a/installperl
+++ b/installperl
@@ -191,6 +191,7 @@ my $installsitearch = "$opts{destdir}$Config{installsitearch}";
 my $installman1dir = "$opts{destdir}$Config{installman1dir}";
 my $man1ext = $Config{man1ext};
 my $libperl = $Config{libperl};
+
 # Shared library and dynamic loading suffixes.
 my $so = $Config{so};
 my $dlext = $Config{dlext};
@@ -563,6 +564,11 @@ sub installlib {
     my $dir = $File::Find::dir;
     $dir =~ s!\Alib/?!!;
 
+    if ($dir =~ m!^\Q$ver\E(?:/|$)!) {    
+        $File::Find::prune = 1;
+        return;
+    }
+
     m!([^/]+)\z!;
     my $name = $1;
 
@@ -659,19 +665,25 @@ sub installlib {
     }
 
     if (-f $_) {
-	my $xname = "$installlib/$name";
-	$xname =~ s/^\Q$opts{destdir}\E//;
+        my $dest = "$installlib/$name";
+        $dest =~ s!(^\Q$installlib\E)/(?:\Q$ver\E/)?!$1/!;
+        my $xname = $dest;
+        $xname =~ s/^\Q$opts{destdir}\E//;
 	$packlist->{$xname} = { type => 'file' };
-        if ($opts{force} || compare($_, "$installlib/$name") || $opts{notify}) {
-	    unlink("$installlib/$name");
-	    mkpath("$installlib/$dir");
+        if ($opts{force} || compare($_, $dest) || $opts{notify}) {    
+            unlink($dest);
+            mkpath("$installlib/$dir");
+            print STDERR "FROM=$_\n";
+            print STDERR "DIR=$dir\n";
+            print STDERR "NAME=$name\n";
+            print STDERR "INSTALLLIB=$installlib\n";
+            print STDERR "DEST=$dest\n\n";
 	    # HP-UX (at least) needs to maintain execute permissions
 	    # on dynamically-loaded libraries.
-            if (copy_if_diff($_, "$installlib/$name")) {
-                strip("-S", "$installlib/$name")
+            if (copy_if_diff($_, $dest)) {    
+                strip("-S", $dest)    
                     if $^O eq 'darwin' and /\.(?:so|$dlext|a)$/;
-                chmod(/\.(so|$dlext)$/ ? $SO_MODE : $NON_SO_MODE,
-                      "$installlib/$name");
+                chmod(/\.(so|$dlext)$/ ? $SO_MODE : $NON_SO_MODE, $dest);
             }
 	}
     }
